### PR TITLE
Hoppscotch UI

### PR DESCRIPTION
**Pull request by anwarulislam**
### Description

this PR separates the UI components from the `hoppscotch-common` package and moves them into a new package called `hoppscotch-ui`. These components are presentational only and include items such as buttons and modals.

Moving these components into a separate package allows us to easily manage and maintain them independently from the rest of the `hoppscotch` codebase. This will also allow us to easily reuse these components across different projects, as they will not be tied to the rest of the `hoppscotch` codebase.

In addition to separating the UI components, we are adding `Histoire` to this PR. That allows us to easily view and interact with our components in isolation, making it easier to develop and test them.

Making sure to thoroughly test these components to ensure that they are working as expected and are ready to be used in other projects.

I've made changes to `ConfirmModal.vue`, `Expand.vue`, `Modal.vue`, and `Windows.vue` components. Specifically, I've removed the i18n functionality and added props for localization. If no props are passed, default English labels will be applied. This change allows for more flexibility and makes it easier to implement localization in different projects.

### Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

**Review by AndrewBastin — CHANGES_REQUESTED** [review]
There are a couple of issues with the current implementation and it should be fixed.

1.`hoppscotch-ui` should either stick to being fully built or not being built but referred to by other packages. Don't try to support both, it is not going to work properly.
2. Certain components like (`AccentModePicker`, `ChangeLanguage` etc.) shouldn't be in `@hoppscotch/ui`, it is fairly obvious why. But anyways, here is a checklist that each component should pass inorder to be considered to be part of `@hoppscotch/ui`:
  - The component is focused towards rendering something. 
  - The component doesn't perform any outside world updates. (Updating outside DOM, run outside functions etc.)
  - The component doesn't import any outside composables/functions (like from `@hoppscotch/common` or anything similar)
  
  Only consider those components for `@hoppscotch/ui`.

File: packages/hoppscotch-ui/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { Plugin } from "vue"
+import * as components from "./components/"
+
+const HoppUI = { ...components }
+
+// Hopp UI will be available globally when in a browser environment
+if (typeof window !== "undefined") {
+  ;(window as any)["HoppUI"] = HoppUI
+}
+
```
Comment (AndrewBastin) [inline packages/hoppscotch-ui/src/index.ts:10]: This breaks tree shaking of components.

File: packages/hoppscotch-ui/.gitignore [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+*.sw?
+
+# Environment Variables
+.env
+
+# Backend Code generation
+src/helpers/backend/graphql.ts
+src/helpers/backend/backend-schema.json
+
+# vite-plugin-pages-sitemap generates its files in the public directory
```
Comment (AndrewBastin) [inline packages/hoppscotch-ui/.gitignore:33]: ???

Only add stuff required to be in .gitignore

File: packages/hoppscotch-ui/.prettierignore [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+.nuxt
+.hoppscotch
+.vscode
+package-lock.json
+node_modules
+dist
+static
+components.d.ts
+src/types
+src/helpers/backend/graphql.ts
```
Comment (AndrewBastin) [inline packages/hoppscotch-ui/.prettierignore:12]: Refine this into only the required items

File: packages/hoppscotch-ui/histoire.config.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+import { FileSystemIconLoader } from "unplugin-icons/loaders"
+import IconResolver from "unplugin-icons/resolver"
+import Icons from "unplugin-icons/vite"
+import Components from "unplugin-vue-components/vite"
+import WindiCSS from "vite-plugin-windicss"
+
+export default defineConfig({
+  setupFile: "histoire.setup.ts",
+  plugins: [HstVue()],
+  vite(config, env) {},
```
Comment (AndrewBastin) [inline packages/hoppscotch-ui/histoire.config.ts:13]: Underscores for unused variables ? (In this case you can just remove it though).

File: packages/hoppscotch-ui/histoire.setup.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import "./src/assets/scss/styles.scss"
+import "./src/assets/scss/themes.scss"
+import "virtual:windi.css"
+
+export function setupVue3({ app }) {
+  app.provide("test", "hello")
+  // app.use(...)
```
Comment (AndrewBastin) [inline packages/hoppscotch-ui/histoire.setup.ts:7]: ??

**Review by AndrewBastin — CHANGES_REQUESTED** [review]
look into the given comments.

File: packages/hoppscotch-ui/README.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+This project is licensed under the [MIT License](https://opensource.org/licenses/MIT) - see [`LICENSE`](https://github.com/hoppscotch/hopp-js-sandbox/blob/main/LICENSE) for more details.
+
+<div  align="center">
+
+<br />
+
+<br />
+
+###### built with ❤︎ by the [Hoppscotch Team](https://github.com/hoppscotch) and [contributors](https://github.com/AndrewBastin/hopp-js-sandbox/graphs/contributors).
+
```
Comment (AndrewBastin) [inline packages/hoppscotch-ui/README.md:60]: Wrong URL for the contributors link

File: packages/hoppscotch-ui/README.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+We've included Histoire in this library which is similar to Storybook, to make it easy to play with the components in the browser. You can run Histoire in the browser with command
+
+`pnpm run story:dev`
+
+You can also use Histoire to create stories for your components and test them in different scenarios.
+
+## Examples
+
+For examples of how to use the components in `hoppscotch-ui`, please refer to the documentation on the [hoppscotch-ui website](https://hoppscotch.io/)
```
Comment (AndrewBastin) [inline packages/hoppscotch-ui/README.md:43]: You can remove the line about documentation as there is no documentation.

File: packages/hoppscotch-ui/README.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+## Usage
+
+To use the components in project, simply name the component with `directory` name as alias:
+
+For example `Primary Button` component is in `button` directory and the file name is `Primary.vue`. So, use that you have to write `<ButtonPrimary />`
+
+## Histoire
+
+We've included Histoire in this library which is similar to Storybook, to make it easy to play with the components in the browser. You can run Histoire in the browser with command
```
Comment (AndrewBastin) [inline packages/hoppscotch-ui/README.md:35]: You can add a link to Histoire here.

File: packages/hoppscotch-ui/src/components/smart/Modal.vue [hunk 9]
```diff
@@ -48,7 +48,7 @@
                 <ButtonSecondary
                   v-if="dimissible"
                   v-tippy="{ theme: 'tooltip', delay: [500, 20] }"
-                  :title="closeText ?? 'Close'"
+                  :title="closeText ?? (t ? t('action.close') : 'Close')"
```
Comment (AndrewBastin) [inline packages/hoppscotch-ui/src/components/smart/Modal.vue:51]:
```suggestion
                  :title="closeText ?? t?.('action.close') ?? 'Close'"
```

File: packages/hoppscotch-ui/src/components/smart/Modal.vue [hunk 10]
```diff
@@ -154,7 +162,7 @@ onMounted(() => {
   stack.push(stackId.value)
   document.addEventListener("keydown", onKeyDown)
 
-  // disableKeybindings()
+  onModalOpen && onModalOpen()
```
Comment (AndrewBastin) [inline packages/hoppscotch-ui/src/components/smart/Modal.vue:165]: See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#optional_chaining_with_function_calls

```suggestion
  onModalOpen?.()
```

File: packages/hoppscotch-ui/src/components/smart/Modal.vue [hunk 11]
```diff
@@ -122,20 +133,17 @@ defineProps({
...
 }>()
 
-// TEMP: Disable keybindings when modal is open
-// const { disableKeybindings, enableKeybindings } = useKeybindingDisabler()
-
-// onBeforeUnmount(() => {
-//   enableKeybindings()
-// })
+onBeforeUnmount(() => {
+  onModalClose && onModalClose()
```
Comment (AndrewBastin) [inline packages/hoppscotch-ui/src/components/smart/Modal.vue:145]:
```suggestion
  onModalClose?.()
```

**Review by AndrewBastin — CHANGES_REQUESTED** [review]
look into the comments

File: packages/hoppscotch-common/src/components/app/Footer.vue [hunk 12]
```diff
@@ -271,7 +270,7 @@ const showDeveloperOptionModal = () => {
 
 // Template refs
 const tippyActions = ref<TippyComponent | null>(null)
-const documentation = ref<typeof SmartItem | null>(null)
-const shortcuts = ref<typeof SmartItem | null>(null)
-const chat = ref<typeof SmartItem | null>(null)
+const documentation = ref<any | null>(null) // any is a hack here. SmartItem type will be here
+const shortcuts = ref<any | null>(null) // any is a hack here. SmartItem type will be here
+const chat = ref<any | null>(null) // any is a hack here. SmartItem type will be here
```
Comment (AndrewBastin) [inline packages/hoppscotch-common/src/components/app/Footer.vue:275]: This doesn't need to be `any`

File: packages/hoppscotch-common/src/components/environments/teams/Environment.vue [hunk 13]
```diff
@@ -126,9 +125,9 @@ const confirmRemove = ref(false)
 
 const tippyActions = ref<TippyComponent | null>(null)
 const options = ref<TippyComponent | null>(null)
-const edit = ref<typeof SmartItem | null>(null)
-const duplicate = ref<typeof SmartItem | null>(null)
-const deleteAction = ref<typeof SmartItem | null>(null)
+const edit = ref<any | null>(null) // any is a hack here. SmartItem type will be here
+const duplicate = ref<any | null>(null) // any is a hack here. SmartItem type will be here
+const deleteAction = ref<any | null>(null) // any is a hack here. SmartItem type will be here
```
Comment (AndrewBastin) [inline packages/hoppscotch-common/src/components/environments/teams/Environment.vue:130]: This doesn't need to be `any`

File: packages/hoppscotch-ui/src/index.ts [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+import "./assets/scss/styles.scss"
+import "virtual:windi.css"
+
+export type HoppUIPluginOptions = {
+  t?: (key: string) => string
+  onModalOpen?: () => void
+  onModalClose?: () => void
+}
```
Comment (AndrewBastin) [inline packages/hoppscotch-ui/src/index.ts:18]: You need to add proper JSDoc comments defining what each field does for public types.

File: packages/hoppscotch-ui/src/components/smart/Modal.vue (hunk 10 above)
Comment (anwarulislam) [reply packages/hoppscotch-ui/src/components/smart/Modal.vue:165]: Thanks. I'll remember this from the next time :D

**Review by AndrewBastin — DISMISSED** [review]
lgtm 💯

**Comment by AndrewBastin** [comment]
@liyasthomas you can perform your review and merge when ready

**Review by liyasthomas — APPROVED** [review]
LGTM

**Review by liyasthomas — APPROVED** [review]

**Pull request merged**